assets/src/components/CustomLink/CustomLink.js

```diff
@@ -0,0 +1,48 @@
+import {Component} from '@wordpress/element';
+import {URLInput} from '@wordpress/block-editor'
+import {TextControl as BaseTextControl} from '@wordpress/components';
+import withCharacterCounter from '../../components/withCharacterCounter/withCharacterCounter';
+
+const TextControl = withCharacterCounter( BaseTextControl );
```

Comment:
Not sure if any url field would require a character counter? The `TextControl` elements that were used for urls in some places technically were enhanced with this function because they didn't have a dedicated element, however it didn't do anything anyway as the props required for the counter were not there.

But can't we just make every instance use `UrlInput`? That would make this new component a lot simpler.